sn_node/src/node/bootstrap/join.rs

```diff
@@ -375,6 +372,43 @@ impl<'a> Joiner<'a> {
         }
     }
 
+    // The retry responses may containing different expected ages, or only having
+    // more than 1/3 of elders copies (which blocks the aggregation).
+    // So a restart with new name shall happen when: received 1/3 of elders copies
+    // And with an age of: the lowest expected age
+    fn insert_retry_response(
+        &mut self,
+        sender: Peer,
+        expected_age: u8,
+        elders: BTreeSet<Peer>,
+        elders_count: u8,
+    ) -> Option<u8> {
+        if !elders.contains(&sender) {
+            error!(
+                "Sender {:?} of the retry-response is not part of the elders {:?}",
+                sender, elders
+            );
+            return None;
+        }
+        let _ = self.retry_responses_cache.insert(sender, expected_age);
+        let mut responded = 0;
+        let mut min_expected_age = 255;
+        for (elder, expected_age) in self.retry_responses_cache.iter() {
+            if elders.contains(elder) {
+                responded += 1;
+                if min_expected_age > *expected_age {
+                    min_expected_age = *expected_age;
+                }
+            }
+        }
+
+        if responded > elders_count / 3 && self.node.age() != expected_age && !self.aggregated {
+            Some(min_expected_age)
+        } else {
+            None
+        }
```

Comment:
Should the node age check be against the `min_expected_age`?
```suggestion
        if responded > elders_count / 3 && self.node.age() != min_expected_age && !self.aggregated {
            Some(min_expected_age)
        } else {
            None
        }
```